pkg/solana/chainwriter/chain_writer_test.go

```diff
@@ -0,0 +1,691 @@
+		txID := uuid.NewString()
+		args := map[string]interface{}{}
+		submitErr := cw.SubmitTransaction(ctx, "6AfuXF6HapDUhQfE4nQG9C1SGtA1YjP3icaJyRfU4RyE", "initializeLookupTable", args, txID, programID.String(), nil, nil)
+		require.Error(t, submitErr)
+	})
+
+	t.Run("fails if invalid contract name provided", func(t *testing.T) {
+		txID := uuid.NewString()
+		args := map[string]interface{}{}
+		submitErr := cw.SubmitTransaction(ctx, "contract_reader_interface", "initializeLookupTable", args, txID, programID.String(), nil, nil)
+		require.Error(t, submitErr)
+	})
+
+	t.Run("fails if invalid method provided", func(t *testing.T) {
+		txID := uuid.NewString()
+		args := map[string]interface{}{}
+		submitErr := cw.SubmitTransaction(ctx, "6AfuXF6HapDUhQfE4nQG9C1SGtA1YjP3icaJyRfU4RyE", "badMethod", args, txID, programID.String(), nil, nil)
+		require.Error(t, submitErr)
+	})
+
+	t.Run("submits transaction successfully", func(t *testing.T) {
+		recentBlockHash := solana.Hash{}
+		rw.On("LatestBlockhash", mock.Anything).Return(&rpc.GetLatestBlockhashResult{Value: &rpc.LatestBlockhashResult{Blockhash: recentBlockHash, LastValidBlockHeight: uint64(100)}}, nil).Once()
+		txID := uuid.NewString()
+		configProgramID := solana.MustPublicKeyFromBase58("6AfuXF6HapDUhQfE4nQG9C1SGtA1YjP3icaJyRfU4RyE")
+
+		txm.On("Enqueue", mock.Anything, account1.String(), mock.MatchedBy(func(tx *solana.Transaction) bool {
+			// match transaction fields to ensure it was built as expected
+			require.Equal(t, recentBlockHash, tx.Message.RecentBlockhash)
+			require.Len(t, tx.Message.Instructions, 1)
+			require.Len(t, tx.Message.AccountKeys, 5)                                                // fee payer + derived accounts
+			require.Equal(t, admin, tx.Message.AccountKeys[0])                                       // fee payer
+			require.Equal(t, account1, tx.Message.AccountKeys[1])                                    // account constant
+			require.Equal(t, account2, tx.Message.AccountKeys[2])                                    // account lookup
+			require.Equal(t, account3, tx.Message.AccountKeys[3])                                    // pda lookup
+			require.Equal(t, configProgramID, tx.Message.AccountKeys[4])                             // instruction program ID
+			require.Len(t, tx.Message.AddressTableLookups, 1)                                        // address table look contains entry
+			require.Equal(t, derivedLookupTablePubkey, tx.Message.AddressTableLookups[0].AccountKey) // address table
+			return true
+		}), &txID).Return(nil).Once()
+
+		args := map[string]interface{}{
+			"lookupTable":  chainwriter.GetRandomPubKey(t).Bytes(),
+			"lookup_table": account2.Bytes(),
+			"seed1":        seed1,
+			"seed2":        seed2,
+		}
+		submitErr := cw.SubmitTransaction(ctx, "6AfuXF6HapDUhQfE4nQG9C1SGtA1YjP3icaJyRfU4RyE", "initializeLookupTable", args, txID, programID.String(), nil, nil)
```

Comment:
Also based on our other conversation, I think we can change this program config key back to a human readable name. And think you'll also need to update this to pass a `toAddress`

Reply:
fixed!